packages/runtime/runtime-definitions/api-report/runtime-definitions.api.md

```diff
@@ -151,6 +151,14 @@ export interface IDataStore {
     trySetAlias(alias: string): Promise<AliasResult>;
 }
 
+// @alpha (undocumented)
+export interface IDataStore2 extends IDataStore {
```

Comment:
This interface looks a bit weird to me.
It feels like it tries to please two sets of completely different customers, who should know very little about each other:
1. A user of the "component" who might want it to be singleton (IDataStore methods)
2. A comonent itself, who needs to initialize itself and attach runtime to context as part of this process.

Maybe other way to say it - no matter how guts are implemented, it feels like it should not use inheritance, but rather there should be two district interfaces (even if implemented by same object).

This goes in pair with comment I put in createRootInstance(), where I think attempt to set alias (and return named object, possibly by loading already created object) should be decoupled (Ok to have helper function) from creation flow.

I'm sure you discussed this approach and maybe there are cons to that, would be great to learn about it